renderdoc/driver/metal/metal_device.cpp

```diff
@@ -471,6 +472,205 @@ WrappedMTLRenderPipelineState *WrappedMTLDevice::newRenderPipelineStateWithDescr
   return wrappedMTLRenderPipelineState;
 }
 
+template <typename SerialiserType>
+bool WrappedMTLDevice::Serialise_newComputePipelineStateWithDescriptor(
+    SerialiserType &ser, WrappedMTLComputePipelineState *pipelineState,
+    RDMTL::ComputePipelineDescriptor &descriptor, MTL::PipelineOption options,
+    MTL::AutoreleasedComputePipelineReflection *reflection, NS::Error **error)
+{
+  SERIALISE_ELEMENT_LOCAL(ComputePipelineState, GetResID(pipelineState))
+      .TypedAs("MTLComputePipelineState"_lit);
+  SERIALISE_ELEMENT(descriptor);
+
+  SERIALISE_CHECK_READ_ERRORS();
+
+  // TODO: implement RD MTL replay
+  if(IsReplayingAndReading())
+  {
+    ResourceId liveID;
+
+    MTL::ComputePipelineDescriptor *mtlDescriptor(descriptor);
+    MTL::ComputePipelineState *realMTLComputePipelineState =
+        Unwrap(this)->newComputePipelineState(mtlDescriptor, options, reflection, error);
+    mtlDescriptor->release();
+    WrappedMTLComputePipelineState *wrappedMTLComputePipelineState;
+    liveID = GetResourceManager()->WrapResource(realMTLComputePipelineState,
+                                                wrappedMTLComputePipelineState);
+    GetResourceManager()->AddLiveResource(ComputePipelineState, wrappedMTLComputePipelineState);
+    AddResource(ComputePipelineState, ResourceType::PipelineState, "Pipeline State");
+    DerivedResource(this, ComputePipelineState);
+  }
+  return true;
+}
+
+WrappedMTLComputePipelineState *WrappedMTLDevice::newComputePipelineStateWithDescriptor(
+    RDMTL::ComputePipelineDescriptor &descriptor, MTL::PipelineOption options,
+    MTL::AutoreleasedComputePipelineReflection *reflection, NS::Error **error)
+{
+  MTL::ComputePipelineDescriptor *realDescriptor(descriptor);
+  MTL::ComputePipelineState *realMTLComputePipelineState;
+  SERIALISE_TIME_CALL(realMTLComputePipelineState = Unwrap(this)->newComputePipelineState(
+                          realDescriptor, options, reflection, error));
+  realDescriptor->release();
+
```

Comment:
Please can we add a TODO here to add error handling for replay if the creation fails